modules/nf-core/shinyngs/staticdifferential/main.nf

```diff
@@ -0,0 +1,41 @@
+process SHINYNGS_STATICDIFFERENTIAL {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda (params.enable_conda ? "bioconda::r-shinyngs=1.4.2" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/r-shinyngs:1.4.2--r41hdfd78af_0':
+        'quay.io/biocontainers/r-shinyngs:1.4.2--r41hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(differential_result)                              // Differential info: contrast and differential stats
+    tuple val(meta2), path(sample), path(feature_meta), path(assay_file)    // Experiment-level info
+
+    output:
+    tuple val(meta), path("*/png/volcano.png"), path("*/html/volcano.html"), path("*/html/volcano_files")   , emit: volcanos
+    path "versions.yml"                                                                                     , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    // For full list of available args see
+    // https://github.com/pinin4fjords/shinyngs/blob/develop/exec/differential_plots.R
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: meta.id
+    """
+    differential_plots.R \\
+        --differential_file $differential_result \\
+        --feature_metadata $feature_meta \\
+        --reference_level $meta.reference \\
+        --treatment_level $meta.target \\
```

Comment:
You should not assume any `meta`, this is up to the pipeline developer to insert via `args` (they may which to call the meta entry differently).

Reply:
Thanks @jfy133 - but this is mostly for consistency with DESeq2, with contrasts derived from a CSV and passed as a meta in this way. https://github.com/nf-core/modules/blob/master/modules/nf-core/deseq2/differential/main.nf

If that needs to change here, it will have to change there too

Reply:
Would renaming it to something else (it was called `contrast_meta` at one point), help distinguish from the more common situations?

Reply:
Basically I was thinking of this as analogous to the sample sheet situation of e.g. the rna-seq workflow, with lists of contrasts rather than lists of samples.

Would you rather the components be flattened into the input channel? If so I'll PR that to the DESeq module too, before we do any more differential work.

Reply:
No, I would just drop them entirely. A user still can store the information within the meta, but they may wish to call the 'entries' within the meta something else. Sorry if I wasn't clear, but basically the information they still store in the `meta`, can still be passed to the module (with those flags) via `ext.args` in the `modules.conf`: `ext.args = { "--reference_level ${meta.foo}" }` etc., so I would drop them entirely here to allow more flexibility for the pipeline developer  (if they are necessary they can be documented in the `meta.yml`).

Reply:
Ahh, gotcha now, that makes sense, thank you.